src/fn/sot457.ts

```diff
@@ -0,0 +1,198 @@
+import type { AnyCircuitElement, PcbSilkscreenPath } from "circuit-json"
+import { z } from "zod"
+import { rectpad } from "../helpers/rectpad"
+import { pillpad } from "src/helpers/pillpad"
+import { silkscreenRef, type SilkscreenRef } from "src/helpers/silkscreenRef"
+
+export const sot457_def = z.object({
+  fn: z.string().default("sot457"),
+  num_pins: z.literal(6).default(6),
+  h: z.string().default("2.5mm"),
+  w: z.string().default("2.7mm"),
+  wave: z.boolean().default(false),
+  pillh: z.string().default("0.45mm"),
+  pillw: z.string().default("1.45mm"),
+  pl: z.string().default("0.8mm"),
+  pw: z.string().default("0.55mm"),
+  p: z.string().default("0.95mm"),
+})
+
+export const sot457_wave = z.object({
+  fn: z.literal("sot457_wave").default("sot457_wave"),
+  num_pins: z.literal(6).default(6),
+  h: z.string().default("3mm"),
+  w: z.string().default("4mm"),
+  wave: z.boolean().default(true),
+  pillr: z.string().default("0.225mm"),
+  pillh: z.string().default("0.45mm"),
+  pillw: z.string().default("1.45mm"),
+  pl: z.string().default("1.45mm"),
+  pw: z.string().default("1.5mm"),
+  p: z.string().default("1.475mm"),
+})
+
+export const sot457 = (
+  raw_params: z.input<typeof sot457_def>,
+): { circuitJson: AnyCircuitElement[]; parameters: any } => {
+  if (raw_params?.wave) {
+    raw_params.fn = "sot457_wave"
+    const parameters = sot457_wave.parse(raw_params)
+    console.log("SOT-457 Wave Parameters:", parameters)
+    return {
+      circuitJson: sot457WithoutParsing(parameters),
+      parameters: parameters,
+    }
+  }
+
+  const parameters = sot457_def.parse(raw_params)
+  return {
+    circuitJson: sot457WithoutParsing(parameters),
+    parameters: parameters,
+  }
+}
+export const getCcwSot457Coords = (parameters: {
+  p: number
+  w: number
+  pn: number
+}) => {
+  const { p, w, pn } = parameters
+
+  if (pn === 1) return { x: -w / 2 - 0.1, y: p }
+  if (pn === 2) return { x: -w / 2 - 0.1, y: 0 }
+  if (pn === 3) return { x: -w / 2 - 0.1, y: -p }
+  if (pn === 4) return { x: w / 2 + 0.1, y: -p }
+  if (pn === 5) return { x: w / 2 + 0.1, y: 0 }
+  if (pn === 6) return { x: w / 2 + 0.1, y: p }
+
+  throw new Error("Invalid pin number")
+}
+
+export const sot457WithoutParsing = (
+  parameters: z.infer<typeof sot457_def>,
+): AnyCircuitElement[] => {
+  const pads: AnyCircuitElement[] = []
+
+  if (parameters.wave) {
+    for (let i = 1; i <= parameters.num_pins; i++) {
+      if (i === 3 || i === 4) {
```

Comment:
this is basically unreadable, find a different way to do this, perhaps with an object e.g.

```
const pinPositions = {
  1: ({ pw, ph }) => pillpad(...)
}
```